controller/search_blackbox_test.go

```diff
@@ -403,6 +401,140 @@ func (s *searchControllerTestSuite) TestSearchFilter() {
 	assert.Equal(s.T(), "specialwordforsearch", r.Attributes[workitem.SystemTitle])
 }
 
+func (s *searchControllerTestSuite) TestSearchByWorkItemTypeGroup() {
+	s.T().Run(http.StatusText(http.StatusOK), func(t *testing.T) {
+		// given
+		fxt := tf.NewTestFixture(t, s.DB, tf.CreateWorkItemEnvironment())
+		svc := testsupport.ServiceAsUser("TestUpdateWI-Service", *fxt.Identities[0])
+		workitemsCtrl := NewWorkitemsController(svc, gormapplication.NewGormDB(s.DB), s.Configuration)
+		// given work items of different types and in different states
+		type testWI struct {
+			Title          string
+			WorkItemTypeID uuid.UUID
+			State          string
+			SpaceID        uuid.UUID
+		}
+		testWIs := []testWI{
+			{"closed feature", workitem.SystemFeature, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+			{"open feature", workitem.SystemFeature, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"closed bug", workitem.SystemBug, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+			{"open bug", workitem.SystemBug, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"open experience", workitem.SystemExperience, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"closed experience", workitem.SystemExperience, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+			{"open task", workitem.SystemTask, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"closed task", workitem.SystemTask, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+			{"open scenario", workitem.SystemScenario, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"closed scenario", workitem.SystemScenario, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+			{"open fundamental", workitem.SystemFundamental, workitem.SystemStateOpen, fxt.Spaces[0].ID},
+			{"closed fundamental", workitem.SystemFundamental, workitem.SystemStateClosed, fxt.Spaces[0].ID},
+		}
+		for _, wi := range testWIs {
+			payload := minimumRequiredCreateWithTypeAndSpace(wi.WorkItemTypeID, wi.SpaceID)
+			payload.Data.Attributes[workitem.SystemTitle] = wi.Title
+			payload.Data.Attributes[workitem.SystemState] = wi.State
+			_, _ = test.CreateWorkitemsCreated(t, svc.Context, svc, workitemsCtrl, wi.SpaceID, &payload)
+		}
+
+		// helper function that checks if the given to be found work item titles
+		// exist in the result list that originate from a search query.
+		checkToBeFound := func(t *testing.T, toBeFound map[string]struct{}, results []*app.WorkItem) {
+			require.Len(t, results, len(toBeFound))
+			for _, wi := range results {
+				title, ok := wi.Attributes[workitem.SystemTitle].(string)
+				require.True(t, ok)
+				_, ok = toBeFound[title]
+				if ok {
+					delete(toBeFound, title)
+				}
+			}
+			require.Empty(t, toBeFound, "not all work items could be found: %+v", toBeFound)
+		}
+
+		// when
+		t.Run("Scenarios", func(t *testing.T) {
+			// given
+			filter := fmt.Sprintf(`
+				{"$AND": [
+					{"`+search.WITGROUP+`": "Scenarios"},
```

Comment:
Should we make constants for type group names in `typegroup.go` and use it elsewhere?

Reply:
They will surely go away once space templates are in and then we will also have to persist those work item type groups in the DB. And then we cannot use those constants either but we have to make a testfixture for them. That's why I thought it's fine for now. Once we have this PR in we can make the testfixture and then get rid of the whole WI creation above also. I did it the complicated way for now.